lib/semirel.gi

```diff
@@ -972,13 +972,14 @@ function(m)
   fi;
 
   #gens:=Set(GeneratorsOfMonoid(semi));
-  gens:=Set(Filtered(GeneratorsOfMonoid(semi), x-> not IsOne(x)));
+  one := One(semi);
```

Comment:
I'd prefer a uniform formatting here: either use ` one:=One(semi);` here or put the spaces on all the other assignments.